src/elf/parser.rs

```diff
@@ -202,6 +202,77 @@ impl<'mmap> Cache<'mmap> {
             .map_err(|_| Error::new(ErrorKind::InvalidInput, "invalid section name"))?;
         Ok(name)
     }
+
+    fn symbol(&mut self, idx: usize) -> Result<&'mmap Elf64_Sym, Error> {
+        let () = self.ensure_symtab()?;
+        // SANITY: The above `ensure_symtab` ensures we have `symtab`
+        //         available.
+        let symtab = self.symtab.as_ref().unwrap();
+        let symbol = symtab.get(idx).ok_or_else(|| {
+            Error::new(
+                ErrorKind::InvalidInput,
+                format!("ELF symbol index ({idx}) out of bounds"),
+            )
+        })?;
+
+        Ok(symbol)
+    }
+
+    /// Find the section of a given name.
+    ///
+    /// This function return the index of the section if found.
+    fn find_section(&mut self, name: &str) -> Result<usize, Error> {
+        let ehdr = self.ensure_ehdr()?;
+        for i in 0..ehdr.e_shnum.into() {
```

Comment:
Fair enough. Let me change that as part of a follow on change, though, please, because the commit in question is really intended to be a mechanical transformation.